src/test/java/org/mifos/integrationtest/cucumber/stepdef/BillPayStepDef.java

```diff
@@ -160,8 +165,14 @@ public void iCallThePaymentNotificationApiExpectedStatusOf(int expectedStatus, S
 
         logger.info("Payment notiifcation response: {}", scenarioScopeState.response);
         JSONObject jsonObject = new JSONObject(scenarioScopeState.response);
-        scenarioScopeState.transactionId = jsonObject.getString("transactionId");
-        assertThat(scenarioScopeState.transactionId.equals("NA")).isFalse();
+        if (!jsonObject.get("transactionId").toString().equals("null")) {
```

Comment:
we had found that in this case the response doesn't contain transaction Id. hence we had to remove these lines. Please check once. ( during validation, the error object doesn't have transaction Id)